units/check_blocked_tasks.go

```diff
@@ -117,10 +129,10 @@ func checkUnmarkedBlockingTasks(t *task.Task, dependencyCaches map[string]task.T
 			"activated_by": t.ActivatedBy,
 			"depends_on":   t.DependsOn,
 		})
-		return 0, errors.Wrapf(err, "checking if dependencies met for task '%s'", t.Id)
+		return errors.Wrapf(err, "checking if dependencies met for task '%s'", t.Id)
 	}
 	if dependenciesMet {
-		return 0, nil
+		return nil
```

Comment:
GitHub doesn't let me comment on unchanged lines but for L144, if this is updating the blocked dependency state for finished tasks to block them in case `UpdateBlockedDependencies` fails to run, should we also call `MarkDependenciesFinished` to indicate that the blocking task is both blocking its dependency children and also finished running?

I'm also fine if we just want to make a separate ticket to investigate this later.

Reply:
Sounds good, I consolidated this comment and the other one into https://jira.mongodb.org/browse/EVG-20855